### ISSUE TITLE
spark-csv package does not work with hive context

### ISSUE THREAD

**Issue by jotterbach** [databricks/spark-csv#140]
I want to read in a csv file from disk using a Hive Context so I can subsequently use window functions. However it seems that the csv file format provided by this package is not compatible with HiveContext `read()` functionality

The following code is successful:
```
from pyspark.sql import SQLContext
sctx = SQLContext(sc)

raw_sdf = sctx.read.format('com.databricks.spark.csv').options(header='true').load('cars.csv')
```

However if I exchange the SQLContext for a HiveContext
```
from pyspark.sql import HiveContext
hctx = HiveContext(sc)

raw_sdf = hctx.read.format('com.databricks.spark.csv').options(header='true').load('cars.csv')
```

The system breaks with the following stacktrace:

>Py4JJavaError: An error occurred while calling o50.load.
>: java.lang.RuntimeException: Failed to load class for data source: com.databricks.spark.csv
>	at scala.sys.package$.error(package.scala:27)
>	at org.apache.spark.sql.sources.ResolvedDataSource$.lookupDataSource(ddl.scala:216)
>	at org.apache.spark.sql.sources.ResolvedDataSource$.apply(ddl.scala:229)
>	at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:114)
>	at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:104)
>	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
>	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
>	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
>	at java.lang.reflect.Method.invoke(Method.java:606)
>	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:231)
>	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:379)
>	at py4j.Gateway.invoke(Gateway.java:259)
>	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:133)
>	at py4j.commands.CallCommand.execute(CallCommand.java:79)
>	at py4j.GatewayConnection.run(GatewayConnection.java:207)
>	at java.lang.Thread.run(Thread.java:745)

It also becomes impossible to read in the file with a SQLContext once you tried to import it with a HiveContext. You need to restart the SparkContext.

**Comment by yhuai** [comment]
What is the spark version? If it is 1.4.0, can you try 1.4.1?

**Comment by jotterbach** [comment]
The first try was 1.4.0 as you correctly guessed. Tried it with 1.4.1 now and it seems to work. Thanks a lot!

**Comment by yhuai** [comment]
Cool. So, I think https://issues.apache.org/jira/browse/SPARK-8368 is the one fixed the problem in Spark 1.4.1.
